hack/generated/Dockerfile

```diff
@@ -1,36 +1,9 @@
-# Build the manager binary
-FROM golang:1.13.15 as builder
-
-WORKDIR /workspace
-
-# Copy the Go Modules manifests
-COPY go.mod go.mod
-COPY go.sum go.sum
-# cache deps before building and copying source so that we don't need to re-download as much
-# and so that source changes don't invalidate our downloaded layer
-RUN go mod download
-
-# Copy the go source
-# COPY . ./
-COPY main.go main.go
-COPY apis/ apis/
-COPY controllers/ controllers/
-COPY pkg/ pkg/
-
-# Build
-# TODO: Use Makefile here -- right now it's awkward to do so because:
-#   1. tools.mk is required for the makefile from the above directory, but Dockerfile can only look in its directory and below.
-#   2. Having Dockerfile here but building it from above could work except that there's another Dockerfile and a .dockerignore
-#      up above that break things. For now we just build by hand
-# RUN make build
-
-# TODO: Do we want CGO_ENALBED=0 and the other options below in the makefile?
-RUN CGO_ENABLED=0 GOOS=linux GOARCH=amd64 GO111MODULE=on go build -a -o k8sinfra-controller main.go
+# Note: This Dockerfile assumes that the binary has been built using the top-level Taskfile.yml
```

Comment:
The Taskfile task has a dependency on the build task so it will rebuild if needed. It might not be 100% perfect (if we have somehow missed an input file) but in CI we will always build from scratch so won't run into any issues there.